src/utils.js

```diff
@@ -430,6 +432,26 @@ export async function getLock(lockId, callback) {
     return lock.acquire(lockId, callback);
 }
 
+/**
+ * Builds proxy agent options based on upload options. Note that the method may return a falsy value, which
+ * indicates that a proxy does not apply.
+ * @param {DirectBinaryUploadOptions} directBinaryUploadOptions Options from which to retrieve information.
+ * @returns {object} Options for either http-proxy-agent or https-proxy-agent.
+ */
+export function getProxyAgentOptions(directBinaryUploadOptions) {
```

Comment:
Could you put this in one of the files dedicated to proxy code? Since this is clearly proxy code - or create a `proxy-utils` file. Because otherwise this file here will become a pot-pourri of methods...

Reply:
@tmathern good feedback, thank you. I moved the function to `http-utils.js` instead.